src/data/mv_polynomial/variables.lean

```diff
@@ -640,6 +640,15 @@ begin
   refl
 end
 
+lemma mv_polynomial.total_degree_finset_sum {ι : Type*} [decidable_eq ι]
+  (s : finset ι) (f : ι → mv_polynomial σ R) :
+  (s.sum f).total_degree ≤ finset.sup s (mv_polynomial.total_degree ∘ f) :=
+finset.induction_on s (zero_le _)
+(λ a s has hind, by rw finset.sum_insert has;
+  exact le_trans (mv_polynomial.total_degree_add _ _)
+    ( by rw [max_le_iff, finset.sup_insert];
+      exact ⟨ le_sup_iff.mpr $ or.inl (le_of_eq rfl) , le_sup_iff.mpr $ or.inr hind ⟩))
```

Comment:
This feels like a nice way to write this proof to me:
```suggestion
lemma total_degree_finset_sum {ι : Type*} (s : finset ι) (f : ι → mv_polynomial σ R) :
  (s.sum f).total_degree ≤ finset.sup s (λ i, (f i).total_degree) :=
begin
  induction s using finset.cons_induction with a s has hind,
  { exact zero_le _ },
  { rw [finset.sum_cons, finset.sup_cons, sup_eq_max],
    exact (mv_polynomial.total_degree_add _ _).trans (max_le_max le_rfl hind), }
end
```
Note that `sup_eq_max` gets rid of the last messy line.

Reply:
Nice proof! Thank you :)